fastlane/lanes/release.rb

```diff
@@ -555,17 +558,29 @@ def commit_version_and_build_files
   )
 end
 
-def create_release_management_pull_request(base_branch:, title:)
+def create_release_management_pull_request(release_version:, base_branch:, title:)
   token = ENV.fetch('GITHUB_TOKEN', nil)
 
   UI.user_error!('Please export a GitHub API token in the environment as GITHUB_TOKEN') if token.nil?
 
-  create_pull_request(
+  pr_url = create_pull_request(
     api_token: token,
     repo: 'wordpress-mobile/WordPress-iOS',
     title:,
     head: Fastlane::Helper::GitHelper.current_git_branch,
     base: base_branch,
     labels: 'Releases'
   )
+
+  # Next, set the milestone for the PR
+  #
+  # The create_pull_request action has a 'milestone' prameter, but it expects the milestone id.
+  # We don't know the id of the milestone, but we can use a different action to set it.
+  #
+  # PR URLs are in the format github.com/org/repo/pull/id
+  pr_number = pr_url.split('/').last
+  update_pull_requests_milestone(pr_numbers: [pr_number], to_milestone: release_version)
```

Comment:
A bit of a misuse of @AliSoftware recent `update_pull_requests_milestone`, but it does the job.

Or, I should say, _should_ do the job. I haven't had a chance to try it yet since my plan is to test these during the ongoing 24.3 cycle.

Reply:
Actually I wouldn't really qualify this as a misuse of that action, because while the primary goal of that new action was to batch-move PRs from one milestone to the next indeed… the reason why I made this action also accept a `pr_numbers:` argument was precisely in anticipation for this kind of use cases of just setting the milestone of one or more fixed list of PRs… so just like you're using it there 👍 

(I'm now glad that I decided to add that `pr_numbers:` argument and anticipated such use case now, even though at the time it could have been considered YAGNI and premature ~optimization~ generalization 😛 )

Reply:
Love it. 

That's great platform engineering. Build unopinionated tools that implemented opinionated defaults. This way, they can be used out of the box but also allow for more advanced/specific/unusual use cases.